src/shogun/distance/EuclideanDistance.h

```diff
@@ -37,6 +37,13 @@ namespace shogun
  * \f[\displaystyle
  *  d({\bf x},{\bf x'})= \sum_{i=0}^{n}|{\bf x_i}-{\bf x'_i}|^2
  * \f]
+ * 
+ * Distance is computed as :  
+ * \sqrt{{\bf x}\cdot {\bf x} - 2{\bf x}\cdot {\bf x'} + {\bf x'}\cdot {\bf x'}}
+ *  
+ * Squared norms for left hand side and right hand side features can be precomputed.
+ * WARNING : Make sure to reset squared norms using reset_squared_norms() when features
+ * or feature matrix are changed.
```

Comment:
We dont know when feature matrix is changed....
For the cases we know, we can (and should) just call reset.
But you are right it is not yet optimal. I leave it to @Saurabh7 to come up with a better solution